vm/src/slots.rs

```diff
@@ -376,3 +381,14 @@ impl PyComparisonOp {
         }
     }
 }
+
+#[pyimpl]
+pub trait Bufferable: PyValue {
```

Comment:
I think the original name `AsBuffer` you used was better. "hashable object" is a term in python(or cpython?) but bufferable is not.

Though "comparable object" is also nothing in python. @coolreader18 any idea about naming?

Reply:
I think AsBuffer is good, "buffer" isn't a verb so it doesn't really make sense as `Verbable` and there's precedence for `AsNoun` as a trait name in Rust, e.g. `AsRawFd`, `AsRef`

Reply:
AsBuffer is good, but the trait function name I changed it to get_buffer as same as CPython, also because it did return a new BufferProtocol. so we should go back to AsBuffer or something like GetBuffer? what we perfer?

Reply:
I personally think we can keep it as `AsBuffer`

Reply:
That sounds like the implementation trait name can be `Buffer` and the desriptor trait name can be `BufferProtocol`. Like,
- AsBuffer(Bufferable) -> BufferProtocol
- BufferProtocol -> Buffer

Reply:
That makes sense, since a trait is inherently a "protocol", so there's no need to specify that stuff that a buffer can do is a protocol; it's redundant

Reply:
That sounds good!